packages/integration-sdk-runtime/src/synchronization/__tests__/index.test.ts

```diff
@@ -581,28 +591,115 @@ describe('shrinkLargeUpload', () => {
             name: 'test3',
             rawData: {
               testRawData: 'test123',
-              testLargeRawData: 'TRUNCATED',
+              testLargeRawData: expect.stringContaining('aaaaaaa'),
               testFinalData: 'test789',
             },
           },
         ],
       },
     ];
 
-    const shrinkResults = shrinkRawData(data);
-
-    expect(shrinkResults.itemsRemoved).toEqual(3);
+    const shrinkResults = shrinkBatchRawData(data, logger);
+    expect(shrinkResults.itemsRemoved).toEqual(2);
     expect(data).toEqual(finalData);
   });
 });
 
+describe('shrinkLargeUploadDueToProps', () => {
```

Comment:
I wanted to verify the bug so I ran this test with your fix commented out. Yeah, it didn't stop 😎